gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DuplicateObservationStatementMapper.java

```diff
@@ -28,29 +28,27 @@ public class DuplicateObservationStatementMapper {
     public static final int CHAR_LIMIT_FOR_TRUNCATION = 50;
     public static final String ELLIPSIS = "...";
 
-    public void mergeDuplicateObservationStatements(RCMRMT030101UKEhrExtract ehrExtract) {
+    public void mergeOrRemoveDuplicateObservationStatements(RCMRMT030101UKEhrExtract ehrExtract) {
         ehrExtract.getComponent()
                 .stream()
                 .map(RCMRMT030101UKComponent::getEhrFolder)
                 .map(RCMRMT030101UKEhrFolder::getComponent)
                 .flatMap(List::stream)
                 .map(RCMRMT030101UKComponent3::getEhrComposition)
-                .forEach(DuplicateObservationStatementMapper::mergeDuplicateObservationStatements);
+                .forEach(DuplicateObservationStatementMapper::mergeOrRemoveDuplicateObservationStatements);
     }
 
-    private static void mergeDuplicateObservationStatements(RCMRMT030101UKEhrComposition ehrComposition) {
+    private static void mergeOrRemoveDuplicateObservationStatements(RCMRMT030101UKEhrComposition ehrComposition) {
         getLinksetsIn(ehrComposition).stream()
                 .map(id -> getLinkedObservationStatement(ehrComposition, id))
                 .filter(Optional::isPresent)
                 .map(Optional::get)
                 .filter(DuplicateObservationStatementMapper::hasSinglePertinentInformation)
-                .filter(DuplicateObservationStatementMapper::isAnnotationTruncated)
                 .filter(DuplicateObservationStatementMapper::areAdditionalFieldsEmpty)
-                .forEach(truncatedObservationStatement ->
-                        findAndMergeNonTruncatedObservationStatementIntoTruncatedObservationStatement(
-                                truncatedObservationStatement,
-                                ehrComposition.getComponent()
-                        ));
+                .forEach(observationStatement ->
+                             findAndMergeNonTruncatedObservationStatementIntoTruncatedObservationStatementOrRemoveDuplicates(
```

Comment:
This is a humungous method name, at the point we're using words like 'and' and 'or' in method names it suggests to me that the method is more than likely not following the principles:

* A function should be small.
* A function should do one and only one thing.

Reply:
yes, agreed, it is quite long but I am trying to reflect and capture what it is doing as I am adding some other functionality to the existing one (based on Adrian's suggestion) and trying to marry them together.